code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/NetworkDaoImpl.java

```diff
@@ -123,4 +131,30 @@ public Network getNetworkForObject(Object object, String networkKind) {
         }
         return systemNetworks.get(0);
     }
+
+    @Override
+    public IpAddress getHostIPForInstanceWHostNetwork(Instance instance) {
```

Comment:
nit: W... I'll never know what W stands for.  Can you just put `With` (I assume you meant that)